src/lib/common/lazy.ts

```diff
@@ -0,0 +1,5 @@
+export function lazy<T>(cb: () => T) {
+	let defaultValue: T;
+
+	return () => (defaultValue ??= cb());
+}
```

Comment:
We should probably move this to sapphire/utilities

Reply:
https://github.com/sapphiredev/utilities/pull/406